testsuite/features/step_definitions/salt_steps.rb

```diff
@@ -659,6 +659,14 @@ def pillar_get(key, minion)
   end
 end
 
+When(/^I enable repositories before installing Salt on this "([^"]*)"$/) do |host|
+  step %(I enable repository "tools_additional_repo" on this "#{host}")
```

Comment:
One question, shouldn't this be something like:

```suggestion
  step %(I enable repository "tools_additional_repo" on this "#{host}" without error control)
```

IIUC, without this, if `tools_additional_repo` repo is not defined in this node, the scenario will fail.

What do you think?

Reply:
In such a case, I prefer it fails immediately: it helps diagnostic the reason for the failure.
Otherwise, it might or might not fail later; should not fail, we would use the wrong Salt without notice; should fail instead, 
it won't point straight to the final reason for this failure (i.e., a missing repository).

Reply:
The problem I see with this, is that you're making the definition of "additional_repo" in sumaform, an particularly a repository called `tools_additional_repo` as mandatory for the SLE minion in any environment/deployment where you want to be able to run the testsuite (since this scenario runs as part "init_client" stage).

I'm not sure we really want this, IMHO the testsuite should be able to run (at least reaching secondary steps) without having to define this particular "additional repo".

If we really want to make this repository mandatory for the testsuite, then I'd rather make it mandatory in "sumaform" and not be defined as an optional "additional repo".

What do you think? :)

Reply:
@calancha please double-check they usages of this step, in case it can break other Cucumber features, for instance features running for QAM/BV.

Reply:
> I'm not sure we really want this, IMHO the testsuite should be able to run (at least reaching secondary steps) without having to define this particular "additional repo".

OK.